hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/records/DoubleWayMultiPathsRecords.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2017 HugeGraph Authors
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.baidu.hugegraph.traversal.algorithm.records;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Stack;
+import java.util.function.Function;
+
+import org.apache.commons.collections.CollectionUtils;
+
+import com.baidu.hugegraph.backend.id.Id;
+import com.baidu.hugegraph.perf.PerfUtil.Watched;
+import com.baidu.hugegraph.traversal.algorithm.HugeTraverser.Path;
+import com.baidu.hugegraph.traversal.algorithm.HugeTraverser.PathSet;
+import com.baidu.hugegraph.traversal.algorithm.records.record.IntIterator;
+import com.baidu.hugegraph.traversal.algorithm.records.record.Record;
+import com.baidu.hugegraph.traversal.algorithm.records.record.RecordType;
+import com.google.common.collect.Lists;
+
+public class DoubleWayMultiPathsRecords extends AbstractRecords {
+
+    protected final Stack<Record> sourceRecords;
+    protected final Stack<Record> targetRecords;
+
+    protected Record currentRecord;
+    private IntIterator lastRecordKeys;
+    protected int current;
+    protected boolean forward;
+    private int accessed;
+
+    public DoubleWayMultiPathsRecords(RecordType type, boolean concurrent,
+                                      Id sourceV, Id targetV) {
+        super(type, concurrent);
```

Comment:
can set fixed RecordType here?

Reply:
shortestpath and paths use different RecordType